initialization/initialization.go

```diff
@@ -66,6 +60,7 @@ type option struct {
 
 	logger            *Logger
 	powDifficultyFunc func(uint64) []byte
+	referenceOracle   *oracle.WorkOracle
```

Comment:
`WorkOracle` doesn't implement `Scrypter`, instead it uses an `Scrypter` (implemented by `postrs.Scrypt`) and extends it with some additional error checking. Otherwise I would use it 🙂